packages/compass-indexes/src/modules/search-indexes.ts

```diff
@@ -17,13 +17,41 @@ import type { FetchReason } from '../utils/fetch-reason';
 import type { IndexesThunkAction } from '.';
 import { switchToSearchIndexes } from './index-view';
 import type { IndexViewChangedAction } from './index-view';
+import semver from 'semver';
 
 const ATLAS_SEARCH_SERVER_ERRORS: Record<string, string> = {
   InvalidIndexSpecificationOption: 'Invalid index definition.',
   IndexAlreadyExists:
     'This index name is already in use. Please choose another one.',
 };
 
+/*const MIN_VERSION_FOR_VIEW_SEARCH_COMPATIBILITY_COMPASS = '8.1.0';
+
+const isVersionSearchCompatibleForViews = (serverVersion: string) => {
+  try {
+    return semver.gte(
+        serverVersion,
+        MIN_VERSION_FOR_VIEW_SEARCH_COMPATIBILITY_COMPASS
+    );
+  } catch {
+    return false;
+  }
+};*/
+
+const MIN_VERSION_FOR_VIEW_SEARCH_COMPATIBILITY_COMPASS = '8.1.0';
+export const compareVersionForViewSearchCompatibility = (
```

Comment:
If (see my other comment) we need so many different comparison methods, I think we should still keep the pattern of having focused, clearly named, check methods instead of a very generic `compare` one, preferably explaining the reason why 🙂 Looking at current code, there is no way at all right now to say why exactly those arbitrary checks are used and this makes it very hard to maintain long term